src/markov/markov_approx.jl

```diff
@@ -219,3 +223,543 @@ function estimate_mc_discrete{T}(X::Vector{T})
+
+doc"""
+
+Compute a finite-state Markov chain approximation to a VAR(1) process of the form
+
+```math
+    y_{t+1} = b + By_{t} + \Psi^{\frac{1}{2}}\epsilon_{t+1}
+```
+
+where ``\epsilon_{t+1}`` is an vector of independent standard normal
+innovations of length `M`
+
+```julia
+P, X = discrete_var(b, B, Psi, Nm, n_moments, method, n_sigmas)
+```
+
+##### Arguments
+
+- `b::Union{Real, AbstractVector}` : constant vector of length `M`.
+                                     `M=1` corresponds scalar case
+- `B::Union{Real, AbstractMatrix}` : `M x M` matrix of impact coefficients
+- `Psi::Union{Real, AbstractMatrix}` : `M x M` variance-covariance matrix of
+                                       the innovations
+    - `discrete_var` only accepts non-singular variance-covariance matrices, `Psi`.
+- `Nm::Integer > 3` : Desired number of discrete points in each dimension
+
+##### Optional
+
+- `n_moments::Integer` : Desired number of moments to match. The default is 2.
+- `method::VAREstimationMethod` : Specify the method used to determine the grid
+                                  points. Accepted inputs are `Even()`.
+                                  Please see the paper for more details.
+                                  NOTE: `Quantile()` and `Quadrature()` are
+                                        not supported now.
+- `n_sigmas::Real` : If the `Even()` option is specified, `n_sigmas` is used to
+                     determine the number of unconditional standard deviations
+                     used to set the endpoints of the grid. The default is
+                     `sqrt(Nm-1)`.
+
+##### Returns
+
+- `P` : `Nm^M x Nm^M` probability transition matrix. Each row
+        corresponds to a discrete conditional probability
+        distribution over the state M-tuples in `X`
+- `X` : `M x Nm^M` matrix of states. Each column corresponds to an
+        M-tuple of values which correspond to the state associated
+        with each row of `P`
+
+##### NOTES
+
+- discrete_var only constructs tensor product grids where each dimension
+  contains the same number of points. For this reason it is recommended
+  that this code not be used for problems of more than about 4 or 5
+  dimensions due to curse of dimensionality issues.
+- Future updates will allow for singular variance-covariance matrices and
+  sparse grid specifications.
+
+"""
+function discrete_var(b::Union{Real, AbstractVector},
+                      B::Union{Real, AbstractMatrix},
+                      Psi::Union{Real, AbstractMatrix},
+                      Nm::Integer,
+                      n_moments::Integer=2,
+                      method::VAREstimationMethod=Even(),
+                      n_sigmas::Real=sqrt(Nm-1))
+
+    M, M_ = size(B, 1), size(B, 2)
+
+    # Check size restrictions on matrices
+    M == M_ || throw(ArgumentError("B must be a scalar or square matrix"))
+    size(b, 2) == 1 || throw(ArgumentError("b must be a column vector"))
```

Comment:
This check is not necessary because we only allow scalar or AbstractVector in the type constraint on the function argument